crypto/src/hash/mod.rs

```diff
@@ -35,13 +52,31 @@ impl Hasher for Blake3_256 {
         blake3::hash(values.as_bytes()).into()
     }
 
+    fn merge_many(data: &[Self::Digest]) -> Self::Digest {
+        let mut hasher = blake3::Hasher::new();
+        for data in data.iter() {
+            hasher.update(data);
+        }
+        hasher.finalize().into()
+    }
+
     fn hash_elements<E: FieldElement>(elements: &[E]) -> Self::Digest {
         let bytes = E::elements_as_bytes(elements);
         blake3::hash(&bytes).into()
     }
 
-    fn hash_fn() -> HashFunction {
-        blake3
+    fn merge_with_int(seed: Self::Digest, value: u64) -> Self::Digest {
+        let mut data = [0; 64];
+        data[..32].copy_from_slice(&seed);
+        data[56..].copy_from_slice(&value.to_le_bytes());
```

Comment:
Ooc, why did we choose to append the u64 value at the end and not use data[40]? Same for sha3.

Reply:
no good reason. fixed!